packages/compass-app-stores/src/stores/instance-store.js

```diff
@@ -251,6 +251,17 @@ store.onActivated = (appRegistry) => {
     store.refreshInstance(appRegistry);
   });
 
+  appRegistry.on('collection-renamed', async ({ from, to }) => {
+    // we must fetch the old collection's metadata before refreshing because refreshing the
+    // collection metadata will remove the old collection from the model.
+    const metadata = await store.fetchCollectionMetadata(from);
```

Comment:
Why are we fetching metadata for an old namespace and not a new one? Is it to avoid refreshing collection before we updated collection tabs?

Reply:
as far as I know, there's no way to wait for `refresh-data` to have refreshed the collections.  If we refresh the collection metadata in the instance store before selecting the collection metadata, it would be non-deterministic as to which collection metadata we get back - the stale collection metadata or the new collection metadata

so the flow is currently:
- fetch the old collection metadata
- let the instance store refresh
- in `tabs.ts`, construct new collection metadata by spreading the replacing the namespace in the old collection metadata

Reply:
So yeah, while you can't await on those events, the main handling for it happens in this store, so if that's the only reason, you can just call and await `refreshInstance` method directly, similar to what the handler for this event is doing (with `refreshNamespace` probably being a better choice here so that we don't need to refresh stats for all namespaces, only for affected one)

I do think though that potentially what you're doing is the way to go just because as we discussed, compass doesn't really expects namespace to ever be changed, so refreshing the namespace will effectively leave a tab open for a non-existent namespace, so maybe better to update state first and then trigger a namespace refresh. That's what I was trying to confirm, I assumed you maybe tried it the other way around already.